specs/signing-and-verification-workflow.md

```diff
@@ -8,19 +8,19 @@ The user wants to sign an OCI artifact and push the signature to a repository.
 
 ### Signing Prerequisites
 
-- User has access to the signing certificate and private key or a remote signing service through a notation plug-in.
+- User has access to the signing certificate and private key or a remote signing service through a notation plugin.
 
 ### Signing Steps
 
 1. **Generate signature:** Using notation CLI or any other compliant signing tool, sign the OCI artifact. The signing tool should follow the following guideline.
     1. Verify that the signing certificate is valid and satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
     1. Verify that the signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
     1. Generate signature.
-        1. Generate signature using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes). Also, as part of this step, the user-defined/supplied custom attributes should be added to the annotations of the signature's descriptor.
-        1. If the user wants to timestamp the signature, obtain an [RFC-3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant timestamp for the signature generated in the previous step. Otherwise, continue to the next step.
+        1. Generate signature envelope using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes). Also, as part of this step, the user-defined/supplied custom attributes should be added to the annotations of the signature's descriptor.
+        1. If signing scheme is [`notary.x509`](./signing-scheme.md/#notaryx509) and timestamp countersignature is demanded, extract the [primitive signature](./signature-specification.md#signature-envelope) (the digital signature computed on payload and signed attributes) from the signature envelope generated in the previous step. Compute hash of the signature (the hash algorithm SHOULD be deduced from signing certificate's public key) and send it to a [RFC-3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant `Timestamp Authority (TSA)` for timestamping. Otherwise, continue to the next step. 
```

Comment:
Contents related to plugin is stated in file `plugin-extensibility.md`.

Reply:
Yes, but it may need more details to explain my questions.

Reply:
These details are already included in `plugin-extensibility.md`, https://github.com/notaryproject/specifications/blob/15806fd9169798c7d8d198e7881bb9c39d999e45/specs/plugin-extensibility.md?plain=1#L214 shows the case for `generate-signature`.
For `generate-envelop`, plugin vendor could pick their own TSA server and embed the timestamp token in the envelope. This process should be controlled by the plugin. Notation will NOT modify the response. If plugin vendor would like users to provide TSA server, they could do so via the `plugin-config`, we could add it in future iteration of the spec.

Reply:
@JeyJeyGao Updated the `plugin-extensibility.md` to include more details.

Reply:
I see. I think the notation CLI should provide a flag for setting up the TSA server in the sign command. However, based on your explanation, this flag doesn't work for the plugin with the `generate-envelope` command.

Reply:
> I see. I think the notation CLI should provide a flag for setting up the TSA server in the sign command. However, based on your explanation, this flag doesn't work for the plugin with the `generate-envelope` command.

Actually, it MAY work based on spec. Notation adds the tsa server URL to the `--plugin-config`, plugin with `generate-envelope` capability can then decide whether to use it or not. It's implementation details for Notation though.

Reply:
You are right. It is too detailed to discuss it here.